lib/wallet/src/Cardano/Wallet/Write/Tx/Sign.hs

```diff
@@ -0,0 +1,261 @@
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Copyright: © 2023 IOHK, 2023 Cardano Foundation
+-- License: Apache-2.0
+--
+-- Module for 'signTx' and signing-related utilities for balancing.
+module Cardano.Wallet.Write.Tx.Sign
+    (
+    -- * Signing transactions
+    -- TODO: Move signTx function here
+
+    -- * Signing-related utilities required for balancing
+      estimateSignedTxSize
+
+    , KeyWitnessCount (..)
+    , estimateKeyWitnessCount
```

Comment:
Just a thought: the `estimateSignedTxSize` and `estimateKeyWitnessCount` functions seem like ideal candidates to place within the `Balance.SizeEstimation` module, or a more general `Balance.Estimation` module, given that `estimateKeyWitnessCount` is not a size estimation, per se.

Reply:
I did very much want `estimateKeyWitnessCount` and `signTx` next to each other. `estimateSignedTx` might make sense in `Write.Tx.SizeEstimation` though. I'd suggest to revisit in another PR regardless.